apps/language_server/lib/language_server/providers/folding_range/helpers.ex

```diff
@@ -0,0 +1,14 @@
+defmodule ElixirLS.LanguageServer.Providers.FoldingRange.Helpers do
+  @moduledoc """
+  """
+  def first_and_last_of_list([]), do: :empty_list
+
+  def first_and_last_of_list(list) when is_list(list) do
+    [head | tail] = list
+    do_falo_list(tail, head)
+  end
+
+  defp do_falo_list([], first), do: {first, first}
+  defp do_falo_list([last | []], first), do: {first, last}
+  defp do_falo_list([_ | tail], first), do: do_falo_list(tail, first)
```

Comment:
1. Agreed! I probably wrote it that way because I'd just written a bunch of stack manipulation.
2. Uh, it shouldn't be flattening the list. Is it?

Regardless, fixed here: 888814a